pkg/virt-api/webhooks/validating-webhook/validating-webhook.go

```diff
@@ -610,14 +610,21 @@ func ValidateVirtualMachineInstanceSpec(field *k8sfield.Path, spec *v1.VirtualMa
 		}
 	}
 
-	// Validate memory size if values are not negative
+	// Validate memory size if values are not negative or too small
 	if spec.Domain.Resources.Requests.Memory().Value() < 0 {
 		causes = append(causes, metav1.StatusCause{
 			Type: metav1.CauseTypeFieldValueInvalid,
 			Message: fmt.Sprintf("%s '%s': must be greater than or equal to 0.", field.Child("domain", "resources", "requests", "memory").String(),
 				spec.Domain.Resources.Requests.Memory()),
 			Field: field.Child("domain", "resources", "requests", "memory").String(),
 		})
+	} else if spec.Domain.Resources.Requests.Memory().Value() > 0 && spec.Domain.Resources.Requests.Memory().Cmp(resource.MustParse("1M")) < 0 {
+		causes = append(causes, metav1.StatusCause{
```

Comment:
Can we combine the two memory size tests into 1, with the compare < 1M only?

Reply:
@slintes My first version was like you said, but later I found that other unit tests in this module do not fill memory size and `spec.Domain.Resources.Requests.Memory == 0`. Basically, 0 is the wrong value, but now it's allowed and tests use it. I did not want to change all the tests and was not sure that 0 is wrong. That's why I added an explicit condition for > 0.

Reply:
Also spec.Domain.Resources.Requests is optional and can be 0.

Reply:
ah ok, got, the default 0 value